tg_bot/modules/weather.py

```diff
@@ -0,0 +1,70 @@
+import pyowm
+from pyowm import timeutils, exceptions
+
+from typing import Optional, List
+from tg_bot import dispatcher, updater, API_WEATHER
+from tg_bot.modules.disable import DisableAbleCommandHandler
+
+from telegram import Message, Chat, Update, Bot
+from telegram.ext import run_async
+
+@run_async
+def weather(bot: Bot, update: Update, args: List[str]):
+    if len(args) >= 1:
+        location = " ".join(args)
+        if location.lower() == bot.first_name.lower():
+            update.effective_message.reply_text("I will keep an eye on both happy and sad times!")
+            bot.send_sticker(update.effective_chat.id, BAN_STICKER)
+            return
+
+        try:
+            owm = pyowm.OWM(API_WEATHER, language='en')
+            observation = owm.weather_at_place(location)
+            theweather = observation.get_weather()
+            getloc = observation.get_location()
+            thelocation = getloc.get_name()
+            temperature = theweather.get_temperature(unit='celsius')['temp']
```

Comment:
youre 100% sure that "temp" is in that dict, no chance of a KeyError? if there is, consider using `.get()`